File: 2022/day01/day01-1.mjs
import * as fs from "node:fs";

const inputLines = fs.readFileSync("input.txt", "utf-8").split(/\r?\n/);

let maxValue = 0;
let currentValue = 0;
inputLines.forEach((line) => {
  if (!line.trim()) {
    console.log("Group value:", currentValue, maxValue);

    maxValue = Math.max(maxValue, currentValue);
    currentValue = 0;
    return;
  }

  currentValue += parseInt(line, 10);
});

console.log("Max value:", maxValue);

File: 2022/day01/day01-1.py
with open('input.txt') as f:
    lines = f.read()

rawGroups = lines.split('\n\n')
groups = [[int(line) for line in line.split('\n')] for line in rawGroups]
maxElf = max(sum(group) for group in groups)

print("Max: ", maxElf)
